### PR TITLE
eliminate unnecessary calls to Public_key.compress

### LINKED ISSUE #3150: Refactor duplicated code idiom to single function on Public_key
This idiom:

```
(Public_key.Compressed.to_base58_check (Public_key.compress public_key))
```

Is repeated too often in our codebase, we should go through and refactor to use `Public_key.to_base58_check_compressed`

### PR DESCRIPTION
**Pull request by ZabelTech**
This eliminates unnecessary calls to `Public_key.compress` in the presence of `Public_key.Compressed.to_base58_check`
Closes #3150

### CODE REVIEW

File: src/app/cli/src/init/client.ml [hunk 1]
```diff
@@ -880,8 +880,7 @@ let dump_keypair =
       Secrets.Keypair.Terminal_stdin.read_exn ~which:"Mina keypair" privkey_path
     in
     printf "Public key: %s\nPrivate key: %s\n"
-      ( kp.public_key |> Public_key.compress
-      |> Public_key.Compressed.to_base58_check )
+      ( kp.public_key |> Public_key.Compressed.to_base58_check )
```
Comment (mrmr1993) [inline src/app/cli/src/init/client.ml:883]: I think you'll want `Public_key.to_base58_check` instead, here and elsewhere.

**Review by mrmr1993 — COMMENTED** [review]
Nice :D Requested a change to reflect the type change, but once that's done this seems ready to merge!

File: src/app/cli/src/init/client.ml (hunk 1 above)
Comment (ZabelTech) [reply src/app/cli/src/init/client.ml:883]: thx!

**Comment by mrmr1993** [comment]
!ci-build-me

**Review by mrmr1993 — APPROVED** [review]
Thanks, this is much cleaner!

**Comment by psteckler** [comment]
The CI is showing a syntax error:
```
[1mFile "src/lib/mina_base/gen/gen.ml", line 68, characters 44-46[0m:

68 |                  Public_key.to_base58_check kp.public_key)

Error: Syntax error: ')' expected

File "src/lib/mina_base/gen/gen.ml", line 66, characters 13-14:

66 |            [ ( "public_key"

  This '(' might be unmatched
```

**Comment by ZabelTech** [comment]
oops, I must admit that I haven't actually build the patch, since the changes seemed easy enough and the README made it sound that it would be a bit more involved to actually compile the source.
It-Should-Work-Now™

**Comment by ZabelTech** [comment]
ping

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by psteckler** [comment]
From CI:
```
File "src/lib/mina_base/gen/gen.ml", line 67, characters 24-50:
--
Error: Unbound value Public_key.to_base58_check```

**Comment by mrmr1993** [comment]
This doesn't currently compile, and is stale. Closing.

**Pull request closed without merging**